### PR TITLE
Issues #678 and #672 related changes in Libraries/Drei

### PR DESCRIPTION
**Pull request by wbooze**
Issues #678 and #672 related changes in Libraries/Drei/base.lisp, Libraries/Drei/views.lisp

Apps/Listener/listener.lisp small adjustements for the listener appearance and word wrap behaviour:
interactor pane in listener.lisp needs a :end-of-line :wrap as action but it also depends on the 
fields :text-margin and/or :margin

I also have some file fix-sbcl.lisp which kinda is missing from master!

### CODE REVIEW

File: Lisp-Dep/fix-sbcl.lisp [hunk 1]
```diff
@@ -0,0 +1,52 @@
+(in-package :cl-user)
+
```
Comment (dkochmanski) [inline Lisp-Dep/fix-sbcl.lisp]: what do you need this file for? we use closer-mop for mop functionality. please remove this file from the PR.

File: Apps/Listener/listener.lisp [hunk 2]
```diff
@@ -164,12 +164,14 @@
...
 (defun print-listener-prompt (stream frame)
   (declare (ignore frame))
-  (with-output-as-presentation (stream *package* 'package :single-box t)
-    (print-package-name stream))
-  (princ "> " stream)
-  (let ((h (- (bounding-rectangle-height (stream-output-history stream))
-              (bounding-rectangle-height (or (pane-viewport stream) stream)))))
-    (scroll-extent stream 0 (max 0 h))))
+  (with-output-as-presentation 
+   (stream *package* 'package :single-box t)
```
Comment (dkochmanski) [inline Apps/Listener/listener.lisp]: please keep optinos argument in the same line as with-output-as-presentation macro (like it was before)

File: Apps/Listener/listener.lisp [hunk 3]
```diff
@@ -164,12 +164,14 @@
...
+   (with-drawing-options (stream :text-face :roman text-size :normal)
+			 (print-package-name stream))
```
Comment (dkochmanski) [inline Apps/Listener/listener.lisp]: this is not well indented since `(print-package-name stream)` is macro's body not functions argument

Comment (dkochmanski) [reply Apps/Listener/listener.lisp]: please explain why this change is necessary

File: Libraries/Drei/base.lisp [hunk 4]
```diff
@@ -275,7 +275,7 @@ constituent character of the line."
            until (end-of-buffer-p mark2)
            while (or (eql object #\Space) (eql object #\Tab))
            do (incf indentation
-                    (if (eql (object-after mark2) #\Tab) tab-width 1))
+                    (if (eql (object-after mark2) #\Tab) tab-width (/ tab-width 8)))
```
Comment (dkochmanski) [inline Libraries/Drei/base.lisp]: what does it mean? if `tab-width` is 4 and character is not a tab we indent by 0.5? I don't understand how this could be right, if character is not a tab then indent is 1 like it was before.

File: Libraries/Drei/base.lisp [hunk 5]
```diff
@@ -300,7 +300,7 @@ characters in the region between offset1 and offset2."
        for i from line-start-offset below offset
        do (incf column (if (eql (buffer-object buffer i) #\Tab)
                            (- tab-width (mod column tab-width))
-                           1))
+                           (/ tab-width 8)))
```
Comment (dkochmanski) [inline Libraries/Drei/base.lisp]: ditto

File: Libraries/Drei/base.lisp [hunk 6]
```diff
@@ -786,7 +786,7 @@ in the region delimited by mark1 and mark2."))
                                               offset
                                               tab-width))
                (count (- tab-width (mod column tab-width))))
-          (delete-buffer-range buffer offset 1)
+          (delete-buffer-range buffer offset count)
           (loop repeat count
```
Comment (dkochmanski) [inline Libraries/Drei/base.lisp]: if this change is necessary please make it a separate commit and explain why it is necessary.

File: Libraries/Drei/views.lisp [hunk 7]
```diff
@@ -81,37 +81,46 @@ If empty, tabs every TAB-WIDTH are assumed.")))
...
   (:documentation "Return the width of a tab character on
 `stream' in device units (most likely pixels).")
   (:method ((stream extended-output-stream) (tabify tabify-mixin))
-    (if (tab-space-count tabify)
-        (* (tab-space-count tabify) (space-width stream tabify))
-        (recorded-tab-width tabify))))
+	   (with-accessors ((tab-space-count tab-space-count) (recorded-tab-width recorded-tab-width) (use-tabs use-tabs)) tabify
+			   (if use-tabs
+			       recorded-tab-width
+				 (* (if tab-space-count tab-space-count 8) (space-width stream tabify))))))
```
Comment (dkochmanski) [inline Libraries/Drei/views.lisp]: iden't here and above are awful, please fix that and explain in a commit why this change is necessary.

File: Libraries/Drei/views.lisp [hunk 8]
```diff
@@ -81,37 +81,46 @@ If empty, tabs every TAB-WIDTH are assumed.")))
...
+					    (if (= 0 width)
+						(* tab-space-count (space-width stream tabify))
+					      (- width (mod x width)))))
+				 
+				 (if use-tabs
+				     (when tab-stops
+				       (let ((next (find-if (lambda (pos) (> pos x)) tab-stops)))
+					 (or (and next (- next x)) (round-up x (tab-width stream tabify)))))
+				   (round-up x
+					     (* (if tab-space-count tab-space-count 8) (space-width stream tabify))))))))
```
Comment (dkochmanski) [inline Libraries/Drei/views.lisp]: ditto

File: Libraries/Drei/views.lisp [hunk 9]
```diff
@@ -81,37 +81,46 @@ If empty, tabs every TAB-WIDTH are assumed.")))
...
-    (setf (tab-stops tabify) 
+	   (with-accessors ((tab-stops tab-stops) (recorded-stream recorded-stream)) tabify
+    (setf tab-stops
 	  (and column-list
-	       (sort (mapcar (lambda (col) (* col (space-width (recorded-stream tabify) tabify)))
-			     column-list) 
-		     #'<)))))
+	       (sort (mapcar (lambda (col) (* col (space-width recorded-stream tabify)))
+			     column-list)
+		     #'<))))))
```
Comment (dkochmanski) [inline Libraries/Drei/views.lisp]: ditto

File: Libraries/Drei/base.lisp (hunk 4 above)
Comment (wbooze) [reply Libraries/Drei/base.lisp]: tab-space-count is 8 for a tab, other than that i don't know how or which other metrics go into the computation (like fonts or whatever), since there are many levels of abstraction like presentations/views/fonts etc..
the only thing we are sure is the tab-space-count. So what is 1 ?  is it 1 character or is it 1 in device units etc...
I thought that (/ tab-width 8) would give a more portable expression or even (/ tab-width tab-space-count) for that matter that's all, if you think it's really not needed or wrong then delete it please

File: Libraries/Drei/base.lisp (hunk 5 above)
Comment (wbooze) [reply Libraries/Drei/base.lisp]: same as above :) sorry

File: Libraries/Drei/base.lisp (hunk 6 above)
Comment (wbooze) [reply Libraries/Drei/base.lisp]: please have a look into the issue #678 (probably  #672 too) , there I already described it.
